tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateStockFeatureContext.php

```diff
@@ -141,34 +141,35 @@ public function assertLastErrorIsInvalidStockLocation(): void
 
     /**
      * @Then product :productReference should have following stock information:
-     *
-     * @param string $productReference
-     * @param TableNode $table
      */
     public function assertStockInformationForDefaultShop(string $productReference, TableNode $table): void
     {
-        $this->assertStockInformation($productReference, $table, $this->getDefaultShopId());
+        $this->assertStockInformation(
+            $productReference,
+            $table,
+            $this->getDefaultShopId()
+        );
     }
 
     /**
      * @Then product :productReference should have following stock information for shops :shopReferences:
-     *
-     * @param string $productReference
-     * @param TableNode $table
      */
-    public function assertStockInformationForShops(string $productReference, string $shopReferences, TableNode $table): void
-    {
+    public function assertStockInformationForShops(
```

Comment:
Because the assertions were first added by `update_stock` tests on stock movements. No reason to move.
`StockMovementHistoryFeatureContext` is only about `stock_movement_history`, and history is not the same business feature. That's an artificial view extracted from stock movements.

Reply:
But both assertion steps use the same CQRS query `GetProductStockMovementHistory` so technically they are the same business feature and domain feature, if not we would have two distinct queries.

Although I agree the feature has evolved, and in `update_stock` the initial intent was maybe a bit different initially but mainly because we didn't have this `GetProductStockMovementHistory` yet. But your new step could be used in `update_stock` by adding the extra column `type` this would allow having one endpoint for this query, which will prevent developers from hesitating on which one they should use and reduce the duplication of code.

However the step from `assertProductLastStockMovementsForDefaultShop` is indeed quite different, even if it relies on the same query the idea behind this is quite different, it just relies on the query as an "opportunity". So maybe the steps testing only last stock movement should stay in this context (because they really centered on stock feature) while the ones testing multiple stock movements should be in the `StockMovementHistoryFeatureContext` What do you think?